jobqueue/src/main/java/com/birbit/android/jobqueue/network/NetworkUtilImpl.java

```diff
@@ -73,10 +75,20 @@ void dispatchNetworkChange(Context context) {
 
     @Override
     public int getNetworkStatus(Context context) {
+        int permState = context.checkCallingOrSelfPermission(Manifest.permission.ACCESS_NETWORK_STATE);
+        if (permState != PackageManager.PERMISSION_GRANTED) {
+            return NetworkUtil.DISCONNECTED;
+        }
+
         if (isDozing(context)) {
             return NetworkUtil.DISCONNECTED;
         }
+
         ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (cm == null) {
```

Comment:
I'm wondering how this returns null if `listenNetworkViaConnectivityManager` did not crash ?
Were you able to reproduce this ?